#934 - Refactored image styling for wide, medium, and portrait.

paboden opened the pull request:
Refactored image styling for wide, medium, and portrait, including cleanup of related Image/caption/credit code to remove lint warnings and improve consistency.

Wide images are set to full width with max-width contained caption, Medium images and captions are contained to max-width, Portrait images and captions are contained to max-width and are set to 50/50 in the container.

Jura commented on stories/Atom/Images/ImageCaptionCredit/ImageCaptionCredit.jsx:
```diff
@@ -14,22 +19,17 @@ export const Images = ({
   return (
     <figure data-viewport="true" className={cls('image-figcaption', `${size}`)}>
-      <div data-viewport="true" className='image-figcaption__cart scale-up'>
+      <div data-viewport="true" className="image-figcaption__cart scale-up">
```
Can you refactor this so we avoid excessive markup? I believe required treatment can be encapsulated in css, without a need for additional <div> and classes. Also, please make sure default "wide" treatment is applied out of the box on `<figure><img ...></figure>` without any additional classes

paboden commented:
Hi @Jura 

I was able to get things a little more simplified. I have it set so `<figure><img><...figure>` works with wide settings by default. I was able to remove some of the unnecessary classes like `image-figcaption` and `image-figcaption__cart`.

While I left the div wrapper around the image in the template, the setup will work without it. The div just allows for decoration/animation classes to be applied to the image separately. 

Should I setup an option for "Apply animations" where the scale and slide-in effects are applied, with the default set to be without that wrapping div, viewport attributes, and animation classes? Should we add that info in the documentation?

Jura reviewed (changes requested):
@paboden , this is good, almost there.
Regarding additional markup to enable animations - please remove them all for default implementation of the component. I mean  that additional `<div>` wrapper for the image source, `.opacity-only .inviewport` class and `data-viewport` attribute in basic, foundation level of implementation. It is worth adding suggested treatment for animation implementation to the documentation, but it has to be a feature, not enforced functionality. 
One small bug - Portrait variant shouldn't have a max-width set at the `<figure>` level, it is only captions section which should be capped at 64rem (for extra big screens)

paboden commented:
Hi @Jura, This is ready for review. I moved the animated wrapping/classes to an option, with the default being off and just the simplest setup in place. I think I understood what you meant for the portrait output. I removed the max-width on the wrapping figure and only applied it to the text.
Let me know what you think
Cheers,
Phil
